### ISSUE TITLE
[poc] Validate that scm-conandata can be replaced by user automation

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#8881]
To automate creating packages from source repositories in a centralized way:
- Do a copy of conanfile + conandata from original source
- They should have scm and scm_to_conandata active
- User automation can replace the scm info
- Then do the export, which should work

cc / @kerstinkeller

**Comment by memsharded** [comment]
The problem is that the "auto" strategy will fail if you are outside of the source repo. We need to think about it.

**Comment by memsharded** [comment]
Hi @KerstinKeller

I have done a test in https://github.com/conan-io/conan/pull/8943/files with a possible approach. As messing with the scm in the conandata.yml doesn't seem a good approach (private data, that can be overwritten by Conan at export time), I have thought that controlling the scm fields could be a good approach:

```python
 import os
        from conans import ConanFile

        class Recipe(ConanFile):
            scm = {"type": "git",
                   "url": os.getenv("USER_EXTERNAL_URL", "auto"),
                   "revision": os.getenv("USER_EXTERNAL_COMMIT", "auto")}
```

That works both when the recipe is in the source repo, without doing anything, as it will be resolved to "auto", and it will capture the commit and url, and to create the package from an external repo knowing the url and commit, it is as straightforward as defining those env-vars. What do you think?

**Comment by KerstinKeller** [comment]
Hi @memsharded, sorry for the late reply.
While I think this works locally, what happens if this recipe is uploaded to the server, and then a user tries to build for their platform with  `--build missing`? Or are environments recorded at export time?

I feel, as soon as the recipe is in the cache, it should not rely on env variables at all?

**Comment by memsharded** [comment]
Hi @KerstinKeller, please check the extended test in #8943, I have added the check that new env-vars values will not break existing exported scm data.

They key is that once the SCM data has been captured in the ``conandata.yml``, the following happens when the conanfile.py recipe is loaded from disk:
```python
conan_data = self._load_data(conanfile_path)
            conanfile.conan_data = conan_data
            if conan_data and '.conan' in conan_data:
                scm_data = conan_data['.conan'].get('scm')
                if scm_data:
                    conanfile.scm.update(scm_data)
```
What is in disk conandata.yml, basically overwrites the current SCM structure. So if the fields got a value at "export" time, that will be the value that will be used, no matter what the env-vars say, as they are overwritten.

**Comment by memsharded** [comment]
Hi @KerstinKeller 

The PR that contains the test that validates this approach has been merged for 1.37, so far this approach seems valid, but please let us know otherwise. Thanks!
